crates/base/src/edge_runtime.rs

```diff
@@ -359,11 +373,13 @@ impl EdgeRuntime {
                 tokio::select! {
                     _ = tokio::time::sleep(Duration::from_millis(worker_timeout_ms)) => {
                         debug!("max duration reached for the worker. terminating the worker. (duration {})", human_elapsed(worker_timeout_ms));
+                        send_event_if_event_manager_available(event_sender, WorkerEvents::TimeLimit(PseudoEvent {}));
```

Comment:
Should we move this into `worker_ctx`? So all events are sent from it?

Reply:
The `tokio::select! {` ?

Reply:
I meant the sending events. Since the result of the worker is sent back to the context, it's better to handle sending events from there.

Makes it easier to refactor in the future.

Reply:
hmm..I thought we are returning the status in the `EdgeCallResult`? https://github.com/supabase/edge-runtime/blob/main/crates/base/src/edge_runtime.rs#L265

Reply:
No yet. That’s one of the refactors I’ve been trying to make but it messes with the response.

Reply:
alright, then lets go with the current approach and then refactor it later.